main/lsp/LSPTypechecker.cc

```diff
@@ -217,12 +217,23 @@ vector<core::FileRef> LSPTypechecker::runFastPath(LSPFileUpdates &updates, Worke
     Timer timeit(config->logger, "fast_path");
     // Replace error queue with one that is owned by this thread.
     gs->errorQueue = make_shared<core::ErrorQueue>(gs->errorQueue->logger, gs->errorQueue->tracer, errorFlusher);
-    auto result = updates.fastPathFilesToTypecheck(*gs, *config);
-    config->logger->debug("Added {} files that were not part of the edit to the update set", result.extraFiles.size());
+
+    std::vector<core::FileRef> toTypecheck;
+    toTypecheck.reserve(updates.fastPathExtraFiles.size() + updates.fastPathChangedFiles.size());
```

Comment:
I don't know if this matters anymore, but the reason why I had `extraFiles` appear before `changedFiles` in the `toTypecheck` list was just so that I could `move(result.extraFiles)` without having to copy each extra file into `toTypecheck` one-by-one.

If we're back to needing that (because we don't have a `vector<FileRef>` anymore, it probably makes sense to put the `changedFiles` first, because those are the files that the user edited.

Maybe none of this matters, because I think the file list will get sorted somewhere down the pipeline anyways.

Reply:
It does get sorted right after we add package files in, but maybe sorting isn't necessary?

Reply:
I take it back, we should definitely sort because then we can use `std::unique` to avoid maintaining the changed files set.